### ISSUE TITLE
Nested object detection

### ISSUE THREAD

**Issue by pkhigh** [tensorflow/models#6607]
I am working on a problem where I have to detect an object and then detect sub parts of the objects.
For example: Finding the face in and image and then finding eyes in the face.
When I train 2 different models, 1 for detecting Face only and other for detecting eyes in the face. I get detection confidence of 0.9+ in all the cases. 
But when I try to detect both face and eyes in the image with a single model then I cannot get a confidence score greater than 0.8+ on various images.
I believe the problem is due to that fact that all the major object detection algorithms treat each object as an independent entity. Therefore it assumes that eyes are kind of creating occlusion on face. That is why confidence is lower.
Is there a way where I can train a model which learns the inherent relations between the presence of different objects and sub-objects?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by pkhigh** [comment]
System information
What is the top-level directory of the model you are using? RETINANET
OS Platform and Distribution: ? Ubuntu 16
Compiler version: ? g++-5
CUDA version: ? 9.2
TensorFlow installed from? conda
cuDNN version: ? 7.5
GPU models (for all devices if they are not all the same): ? 1 Nvidia GPU
python --version output: ? 3.6
Exact command to reproduce ? Trained the model on a custom dataset.

**Comment by rootkitchao** [comment]
If the input image resolution is high enough, the effect of using multiple models will be better.The current object detection framework is much less accurate for small objects than large objects.If you want to consider the relationship between objects in the detector, you need to add a phase to the detector or process it in the output.

**Comment by pkhigh** [comment]
How can I add phase to the detector?
I mean there has to some architecture changes and loss function tweaking as well.
Can you give me a head start in creating an object detection pipeline which works well on nested objects(objects inside objects) using a single model.
At present I run 2 separate models. One for the bigger object and other for the sub-parts of that object.
I want to combine this into a single model.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be an old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
